### PR TITLE
.Net: Python: CJK support for text splitter

### PR DESCRIPTION
**Pull request by marlenezw**
### Motivation and Context
The text splitting algorithm has hardcoded values for punctuation based on ASCII punctuation (,.; etc.) Chinese and Japanese use both full-width Unicode punctuation values https://en.wikipedia.org/wiki/Halfwidth_and_Fullwidth_Forms_(Unicode_block) or ideographic punctuation.
If you give the text splitter Japanese text it will incorrectly split in the middle of sentences
The current algorithm defaults to an approximation token counter, which is partly accurate for Indo-European/Latin Alphabet languages but totally out for CJK. This tests uses the BPE token length for cl100k_base (GPT3/4)

This is the Python version of [this PR](https://github.com/microsoft/semantic-kernel/pull/5489)

### Description

This change adds some basic full-width characters and ideographic punctuation to the built-in list.
It also adds a test which checks that it splits a sentence correctly. The first part of the test string (田中の猫はかわいいですね.) is 16 tokens for cl100k_base.

### Contribution Checklist

 The code builds clean without any errors or warnings
The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
All unit tests pass, and I have added new tests where possible
I didn't break anything!

### CODE REVIEW

File: python/pyproject.toml [hunk 1]
```diff
@@ -32,6 +32,8 @@ pydantic = "^2"
 motor = "^3.3.2"
 defusedxml = "^0.7.1"
 pybars4 = "^0.9.13"
+jinja2 = "^3.1.3"
```
Comment (tonybaloney) [inline python/pyproject.toml:35]: Move these to the dev dependencies group below so they aren't dependencies of the whole project

File: python/semantic_kernel/text/text_chunker.py [hunk 2]
```diff
@@ -14,10 +14,14 @@
 TEXT_SPLIT_OPTIONS = [
     ["\n", "\r"],
     ["."],
+    ["。"],
+    ["．"],
     ["?", "!"],
     [";"],
     [":"],
     [","],
```
Comment (tonybaloney) [inline python/semantic_kernel/text/text_chunker.py:22]: Put all the commas in a single list, like the line below with the brackets

File: python/semantic_kernel/text/text_chunker.py [hunk 3]
```diff
@@ -14,10 +14,14 @@
 TEXT_SPLIT_OPTIONS = [
     ["\n", "\r"],
     ["."],
```
Comment (tonybaloney) [inline python/semantic_kernel/text/text_chunker.py:16]: Put all the fullstops in a single list, like the line below with the brackets

File: python/semantic_kernel/text/text_chunker.py [hunk 4]
```diff
@@ -26,10 +30,14 @@
 
 MD_SPLIT_OPTIONS = [
     ["."],
+    ["。"],
```
Comment (tonybaloney) [inline python/semantic_kernel/text/text_chunker.py:33]: As above

File: python/semantic_kernel/text/text_chunker.py [hunk 5]
```diff
@@ -26,10 +30,14 @@
...
+    ["．"],
     ["?", "!"],
     [";"],
     [":"],
     [","],
+    ["，"],
```
Comment (tonybaloney) [inline python/semantic_kernel/text/text_chunker.py:39]: As above

File: python/semantic_kernel/text/text_chunker.py [hunk 6]
```diff
@@ -44,7 +52,7 @@ def _token_counter(text: str) -> int:
 
     TODO: chunking methods should be configurable to allow for different
```
Comment (tonybaloney) [inline python/semantic_kernel/text/text_chunker.py:53]: Note, this is configurable so I don't think this TODO is valid. We should raise a separate issue

**Comment by tonybaloney** [comment]
Related #5490

File: python/pyproject.toml (hunk 1 above)
Comment (marlenezw) [reply python/pyproject.toml:35]: Sounds good! I'm not the one that added Jinja, so hopefully tests still pass!

File: python/semantic_kernel/text/text_chunker.py (hunk 6 above)
Comment (marlenezw) [reply python/semantic_kernel/text/text_chunker.py:53]: Sounds good, will do!

**Comment by marlenezw** [comment]
Ooof, I tried to rebase to fix the merge conflicts with the poetry file and it turned into a bit of a mess. @tonybaloney I'm going to close this PR and re-do it so it doesn't take too much time 😅

**Pull request closed without merging**